installers/rpm-code/codes/amrex.sh

```diff
@@ -2,9 +2,13 @@
 
 amrex_main() {
     codes_dependencies common
-    codes_download https://github.com/AMReX-Codes/amrex/releases/download/22.09/amrex-22.09.tar.gz amrex
-    ./configure \
-        --prefix="${codes_dir[prefix]}"
-    codes_make
-    codes_make_install
+    codes_download https://github.com/AMReX-Codes/amrex/releases/download/24.05/amrex-24.05.tar.gz amrex
+    codes_cmake2   \
+      -DCMAKE_INSTALL_PREFIX="${codes_dir[prefix]}"  \
+      -DAMReX_BUILD_SHARED_LIBS=ON  \
+      -DAMReX_LINEAR_SOLVERS=ON     \
+      -DAMReX_OMP=ON                \
+      -DAMReX_PARTICLES=ON          \
+      -DAMReX_SPACEDIM="1;2;3"
```

Comment:
Probably 3D is sufficient for now as well, assuming that only ImpactX depends on AMReX for now.
```suggestion
      -DAMReX_SPACEDIM="3"
```

Reply:
Let's leave all three so the WarpX package cann build 1D/2D/RZ/3D from it.

Reply:
I think we'll go with all dimensions for now so users have access if they need it. Amrex is deployed across our systems (ex on [sirepo.com/jupyter](sirepo.com/jupyter)) so people may use it outside of impactx.